services/provider/server/server.go

```diff
@@ -206,3 +217,101 @@ func Start(port int, providerServer *OCSProviderServer, opts []grpc.ServerOption
 		klog.Fatalf("failed to start gRPC server: %v", err)
 	}
 }
+
+func (s *OCSProviderServer) getJSONConnectionDetails(consumerResource *ocsv1alpha1.StorageConsumer) ([]externalSC.ExternalResource, error) {
+	var j []externalSC.ExternalResource
+
+	// Configmap with mon endpoints
+	configmap, err := s.consumerAPI.GetConfigMap(monConfigMap, s.namespace)
+	if err != nil {
+		return []externalSC.ExternalResource{}, fmt.Errorf("failed to get %s configMap. %v", monConfigMap, err)
+	}
+
+	// This is done to get address of first mon from the monConfigMap configmap
+	cmData := strings.Split(configmap.Data["data"], ",")
+	// Address of first mon
+	monAddr := cmData[0]
```

Comment:
What if the CM data is empty?

Reply:
I thought we assume clusters with these kinds of resources will be available but anyway added the conditions to check.

Reply:
The above will be a panic if the `cmData` is an empty slice.  please check length/size instead